Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -452,6 +454,8 @@ protected virtual void OnDestroy()
 #endif
         }
 
+        void PerformInputSystemUpdateBeforeRender() => InputSystem.Update(InputUpdateType.BeforeRender);
```

Comment:
This will cause the input system to step twice each frame. This isn't the way to fix the issue. The `TrackedPoseDriver.OnBeforeRender` method is correctly called when running a VR app, so this change would cause a double step and break any user that set the Edit > Project Settings > Input System Package > Update Mode to Manual.

What I think is happening is that the `HandheldARInputDevice` doesn't have `updateBeforeRender` set to true in the input control layout, so I'm guessing maybe the device doesn't have `UpdateBeforeRender` flag set. The main `InputManager.OnUpdate` method loop that is invoked by low level code has an early return if the `updateType` is not part of the `m_UpdateMask`. See https://github.com/Unity-Technologies/InputSystem/blob/c55cdd9f94422b4917045999a1c7b00f52e9bf1f/Packages/com.unity.inputsystem/InputSystem/InputManager.cs#L2857

The `m_UpdateMask` is set to add the `InputUpdateType.BeforeRender` ONLY if an added device has that set.
```csharp
if (device.updateBeforeRender)
    updateMask |= InputUpdateType.BeforeRender;
```
See https://github.com/Unity-Technologies/InputSystem/blob/c55cdd9f94422b4917045999a1c7b00f52e9bf1f/Packages/com.unity.inputsystem/InputSystem/InputManager.cs#LL1220C43-L1220C43

So my guess is that in the user's project, there aren't any added devices with that before render flag set, so then the callback in `TrackedPoseDriver` is never getting invoked when `InputState.currentUpdateType == InputUpdateType.BeforeRender`.

We have code in XRI where we have an `XRSimulatedHMD` where we set the attribute.
```csharp
[InputControlLayout(stateType = typeof(XRSimulatedHMDState), isGenericTypeOfDevice = false, displayName = "XR Simulated HMD", updateBeforeRender = true)]
[Preserve]
public class XRSimulatedHMD : XRHMD
```
See https://github.com/Unity-Technologies/com.unity.xr.interaction.toolkit/blob/80d057af71a66e9b32db863e9fc26f18e041b688/com.unity.xr.interaction.toolkit/Runtime/Inputs/Simulation/XRSimulatedHMD.cs#L12-L14

We might be able to just add that `updateBeforeRender = true` to `HandheldARInputDevice` in com.unity.xr.arsubsystems. Can we try that, or verify that the added device of that layout actually gets the flag set? OpenXR HMD devices get it added for instance.
![UpdateBeforeRender flags](https://github.com/Unity-Technologies/InputSystem/assets/67029035/ac566ce3-6b22-4179-b52c-73c7c257e2d4)